### ISSUE TITLE
FailIf example does not work

### ISSUE THREAD

**Issue by herao67** [amantinband/error-or#109]
Hi, I just tried to use a more complex workflow and used the FailIf example (https://github.com/amantinband/error-or?tab=readme-ov-file#failure)

```
.FailIf(val => val > 2, Error.Validation(description: $"{val} is too big")
```
But the `val` to build the description is unknown (Cannot resolve val). An overload for `FailIf` with a `Func<TValue,Error>` as second parameter would be highly appreciated ;)

The full example:

```csharp
ErrorOr<string> foo = await "5".ToErrorOr()
  .Then(int.Parse)
  .FailIf(val => val > 2, Error.Validation(description: $"{val} is too big")) // Cannot resolve val
  .ThenDoAsync(Task.Delay)
  .ThenDo(val => Console.WriteLine($"Finished waiting {val} milliseconds."))
  .ThenAsync(val => Task.FromResult(val * 2))
  .Then(val => $"The result is {val}")
  .Else(errors => Error.Unexpected(description: "Yikes"))
  .MatchFirst(
      value => value,
      firstError => $"An error occurred: {firstError.Description}");
```
Nevertheless, congratulations to a fantastic library, I love using it!

**Comment by ahmtsen** [comment]
created a PR for this issue @amantinband

**Comment by NKnusperer** [comment]
Will this be merged?

**Comment by amantinband** [comment]
Sorry was super busy with other things forgot about this completely.. merged. Will soon publish a new version

**Comment by amantinband** [comment]
available as part of 2.1.0
